daemon/daemon.go

```diff
@@ -939,7 +939,17 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 	// the graphdrivers it should be safe to do here.
 	_, doStorageMigration := os.LookupEnv("BALENA_MIGRATE_OVERLAY")
 	if config.GraphDriver == "overlay2" && doStorageMigration {
-		logrus.Info("Starting storage migration from aufs to overlay2")
+		if storagemigration.CheckAufsRootExists(config.Root) != nil &&
+			storagemigration.CheckOverlayRootExists(config.Root) != nil {
+			// if both roots exist, assume migration succeeded during a previous run
+			// and commit (dropping aufs data)
+			logrus.Infof("Storage migration commit")
+			if err := storagemigration.Commit(config.Root); err != nil {
+				return nil, errors.Wrap(err, "failed to commit storage migration")
+			}
+		}
+
+		logrus.Info("Storage migration from aufs to overlay2 starting")
 		start := time.Now()
```

Comment:
hmm. depends on what we want to time there I guess, the commit is technically not part of the migration as it will run the next time the engine starts up

Reply:
cc @alexgg wdyt?

Reply:
@lmbarros I just pushed 15f486b4d874142605b31bcde345519da273ae11 which should make control flow more obvious and also avoids running the migration code again after commit completes.